pkg/scalers/cassandra_scaler.go

```diff
@@ -73,117 +86,41 @@ func NewCassandraScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
-	meta := &CassandraMetadata{}
-	var err error
-
-	if val, ok := config.TriggerMetadata["query"]; ok {
-		meta.query = val
-	} else {
-		return nil, fmt.Errorf("no query given")
-	}
-
-	if val, ok := config.TriggerMetadata["targetQueryValue"]; ok {
-		targetQueryValue, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, fmt.Errorf("targetQueryValue parsing error %w", err)
-		}
-		meta.targetQueryValue = targetQueryValue
-	} else {
-		if config.AsMetricSource {
-			meta.targetQueryValue = 0
-		} else {
-			return nil, fmt.Errorf("no targetQueryValue given")
-		}
-	}
-
-	meta.activationTargetQueryValue = 0
-	if val, ok := config.TriggerMetadata["activationTargetQueryValue"]; ok {
-		activationTargetQueryValue, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, fmt.Errorf("activationTargetQueryValue parsing error %w", err)
-		}
-		meta.activationTargetQueryValue = activationTargetQueryValue
-	}
-
-	if val, ok := config.TriggerMetadata["username"]; ok {
-		meta.username = val
-	} else {
-		return nil, fmt.Errorf("no username given")
-	}
-
-	if val, ok := config.TriggerMetadata["port"]; ok {
-		port, err := strconv.Atoi(val)
-		if err != nil {
-			return nil, fmt.Errorf("port parsing error %w", err)
-		}
-		meta.port = port
-	}
-
-	if val, ok := config.TriggerMetadata["clusterIPAddress"]; ok {
-		splitval := strings.Split(val, ":")
-		port := splitval[len(splitval)-1]
-
-		_, err := strconv.Atoi(port)
-		switch {
-		case err == nil:
-			meta.clusterIPAddress = val
-		case meta.port > 0:
-			meta.clusterIPAddress = net.JoinHostPort(val, fmt.Sprintf("%d", meta.port))
-		default:
-			return nil, fmt.Errorf("no port given")
-		}
-	} else {
-		return nil, fmt.Errorf("no cluster IP address given")
-	}
-
-	if val, ok := config.TriggerMetadata["protocolVersion"]; ok {
-		protocolVersion, err := strconv.Atoi(val)
-		if err != nil {
-			return nil, fmt.Errorf("protocolVersion parsing error %w", err)
-		}
-		meta.protocolVersion = protocolVersion
-	} else {
-		meta.protocolVersion = 4
+func parseCassandraMetadata(config *scalersconfig.ScalerConfig) (cassandraMetadata, error) {
+	meta := cassandraMetadata{}
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, fmt.Errorf("error parsing cassandra metadata: %w", err)
 	}
 
-	if val, ok := config.TriggerMetadata["consistency"]; ok {
-		meta.consistency = gocql.ParseConsistency(val)
-	} else {
-		meta.consistency = gocql.One
+	if config.AsMetricSource {
+		meta.TargetQueryValue = 0
 	}
 
-	if val, ok := config.TriggerMetadata["keyspace"]; ok {
-		meta.keyspace = val
-	} else {
-		return nil, fmt.Errorf("no keyspace given")
-	}
-	if val, ok := config.AuthParams["password"]; ok {
-		meta.password = val
-	} else {
-		return nil, fmt.Errorf("no password given")
+	err = meta.Validate()
+	if err != nil {
+		return meta, err
 	}
 
-	if err = parseCassandraTLS(config, meta); err != nil {
+	err = parseCassandraTLS(&meta)
```

Comment:
Maybe this function can be moved to Validate() as well.

Reply:
I moved the check which is in the `parseCassandraTLS`moved to `Validate()`